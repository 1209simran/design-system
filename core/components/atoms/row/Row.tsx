import * as React from 'react';
import classNames from 'classnames';
import { BaseProps, extractBaseProps } from '@/utils/types';

export interface RowProps extends BaseProps {
  children?: React.ReactNode;
}

export const Row = (props: RowProps) => {
  const { className, children } = props;
  const baseProps = extractBaseProps(props);

  const classes = classNames({
    Row: true,
    [`${className}`]: className
  });
  return <div {...baseProps} className={classes}>{children}</div>;
};

Row.displayName = 'Row';

export default Row;
