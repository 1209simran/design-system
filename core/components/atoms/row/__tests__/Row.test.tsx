import * as React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Row from '../Row';

describe('Row component', () => {
  it('should match the snapshot without RowColumns', () => {
    const tree = shallow(<Row>Row</Row>);
    expect(tree).toMatchSnapshot();
  });
});
describe('Row component', () => {

  const children = <div>Masala Design System</div>;

  it('renders children', () => {
    const { getByTestId } = render(<Row>{children}</Row>);
    expect(getByTestId('DesignSystem-Row')).toContainHTML('<div>Masala Design System</div>');
  });

  describe('Row Component with overwrite class', () => {

    it('overwrite Row class', () => {
      const { getByTestId } = render(<Row className="RowClass">{children}</Row>);
      expect(getByTestId('DesignSystem-Row')).toHaveClass('RowClass');
    });
  });

});
