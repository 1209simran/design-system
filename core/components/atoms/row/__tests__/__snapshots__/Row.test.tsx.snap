// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component should match the snapshot without RowColumns 1`] = `
<div
  className="Row"
  data-test="DesignSystem-Row"
>
  Row
</div>
`;
